### ISSUE TITLE
setGameClient does not consistently work

### ISSUE THREAD

**Issue by zach-morris** [xbmc/xbmc#23042]
### Describe the bug
Here is a clear and concise description of what the problem is:
With Kodi v20, new infosetters are used, and setInfo is being depreciated.  When attempting to play a game using Retroplayer, setting the game client via setGameClient does not work as expected.  Instead, the game file is attempted to play with the video player and fails.  Here are two examples:

##V19 functionality:
```
xbmc.log(msg='zach_tester:  Test started.', level=xbmc.LOGDEBUG)
game_path = 'pacman.zip'  #Enter local path to game file
li = xbmcgui.ListItem(label='Test1 Label',label2='Test1 Label 2',offscreen=True)
info_dict = {'title':'Packman','platform':'Arcade','genres':['Platformer'],'publisher':'Namco','overview':'This is just a test','year':2023,'gameclient':'game.libretro.mame2003_plus'}
	
# v19 functionality
li.setInfo('game',info_dict)
li.setPath(path=game_path)
xbmc.log(msg='zach_tester:  Starting Playback', level=xbmc.LOGDEBUG)
xbmc.log(msg='zach_tester:  Game client is {}'.format(info_dict.get('gameclient')), level=xbmc.LOGDEBUG)
xbmc.Player().play(item=game_path,listitem=li)
xbmc.log(msg='zach_tester:  Test completed', level=xbmc.LOGDEBUG)
```

Results in:
```
zach_tester:  Test started.
Setting game properties through ListItem.setInfo() is deprecated and might be removed in future Kodi versions. Please use the respective setter in InfoTagGame.
zach_tester:  Starting Playback
zach_tester:  Game client is game.libretro.mame2003_plus
zach_tester:  Test completed
RetroPlayer[PLAYER]: Opening: pacman.zip (game plays as expected)
```

##V20 equivalent funtionality:
```
xbmc.log(msg='zach_tester:  Test started.', level=xbmc.LOGDEBUG)
game_path = 'pacman.zip'  #Enter local path to game file
li = xbmcgui.ListItem(label='Test1 Label',label2='Test1 Label 2',offscreen=True)
info_dict = {'title':'Packman','platform':'Arcade','genres':['Platformer'],'publisher':'Namco','overview':'This is just a test','year':2023,'gameclient':'game.libretro.mame2003_plus'}

# v20 functionality
gameinfo = li.getGameInfoTag()
gameinfo.setTitle(info_dict.get('title'))
gameinfo.setPlatform(info_dict.get('platform'))
gameinfo.setGenres(info_dict.get('genres'))
gameinfo.setPublisher(info_dict.get('publisher'))
gameinfo.setOverview(info_dict.get('overview'))
gameinfo.setYear(info_dict.get('year'))
gameinfo.setGameClient(info_dict.get('gameclient'))
xbmc.log(msg='zach_tester:  Starting Playback', level=xbmc.LOGDEBUG)
xbmc.log(msg='zach_tester:  Game client is {}'.format(gameinfo.getGameClient()), level=xbmc.LOGDEBUG)
xbmc.Player().play(item=game_path,listitem=li)
xbmc.log(msg='zach_tester:  Test completed', level=xbmc.LOGDEBUG)
```
Results in:
```
zach_tester:  Test started.
zach_tester:  Starting Playback
zach_tester:  Game client is game.libretro.mame2003_plus
zach_tester:  Test completed
VideoPlayer::OpenFile: pacman.zip (fails to play here)
```

I've also noticed in the documentation that the [Player class documentation](https://alwinesch.github.io/group__python___player.html) does not include getGameInfoTag, but does include getVideoInfoTag, getMusicInfoTag.  Unsure if this is related.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

setGame* works as expected (similar to v19 setInfo)

## Actual Behavior

setGameClient does not work the same as v19 setInfo

### To Reproduce

1. See example above.  I can provide an example addon if necessary.

### Debuglog

The debuglog can be found here:
See log snippets above.

### Screenshots 
Here are some links or screenshots to help explain the problem:

N/A

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

https://forum.kodi.tv/showthread.php?tid=372450&pid=3147728#pid3147728

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [X] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version:
```
Starting Kodi (20.1 (20.1.0) Git:20230312-289ec664e3). Platform: macOS x86 64-bit
Using Release Kodi x64
Kodi compiled 2023-03-12 by Clang 12.0.0 (clang-1200.0.32.29) for macOS x86 64-bit version 10.13 (101300)
Running on Apple Inc. MacBookPro12,1 with macOS 12.6.3 (Version 12.6.3 (Build 21G419)), kernel: Darwin x86 64-bit version 21.6.0
```

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by zach-morris** [comment]
I did notice in the [listitem](https://github.com/xbmc/xbmc/blob/master/xbmc/interfaces/legacy/ListItem.cpp#L707) gameclient is all lowercase, while in [InfoTagGame](https://github.com/xbmc/xbmc/blob/master/xbmc/interfaces/legacy/InfoTagGame.h#L367) it's gameClient

**Comment by garbear** [comment]
> I've also noticed in the documentation that the [Player class documentation](https://alwinesch.github.io/group__python___player.html) does not include getGameInfoTag, but does include getVideoInfoTag, getMusicInfoTag. Unsure if this is related.

Fixed in https://github.com/xbmc/xbmc/pull/23054

**Comment by garbear** [comment]
I've sent a PR to fix this issue: https://github.com/xbmc/xbmc/pull/23055

Can you give that a try? I'm creating testing builds now.

**Comment by zach-morris** [comment]
I'll try the test builds.  Am i reading this correctly that https://github.com/xbmc/xbmc/pull/23055 will set all info tags?  If thats the case, https://github.com/xbmc/xbmc/issues/22752 may also be fixed by thiss

**Comment by garbear** [comment]
Test builds are up.

The idea is that you set the info tag explicitly:

```python
# v20 functionality
gameinfo = li.getGameInfoTag()
gameinfo.setTitle(info_dict.get('title'))
gameinfo.setPlatform(info_dict.get('platform'))
gameinfo.setGenres(info_dict.get('genres'))
gameinfo.setPublisher(info_dict.get('publisher'))
gameinfo.setOverview(info_dict.get('overview'))
gameinfo.setYear(info_dict.get('year'))
gameinfo.setGameClient(info_dict.get('gameclient'))
li.setGameInfoTag(gameinfo)
```

**Comment by zach-morris** [comment]
Currently getting an error:
```
gameinfo = li.getGameInfoTag()
gameinfo.setTitle(info_dict.get('title'))
gameinfo.setPlatform(info_dict.get('platform'))
gameinfo.setGenres(info_dict.get('genres'))
gameinfo.setPublisher(info_dict.get('publisher'))
gameinfo.setOverview(info_dict.get('overview'))
gameinfo.setYear(info_dict.get('year'))
gameinfo.setGameClient(info_dict.get('gameclient'))
li.setGameInfoTag(gameinfo)
xbmc.log(msg='zach_tester:  Starting Playback', level=xbmc.LOGDEBUG)
xbmc.Player().play(item=game_path,listitem=li)
```

Results in:
```
TypeError: Incorrect type passed to "setGameInfoTag", was expecting a "p.xbmc::InfoTagGame" but received a "p.XBMCAddon::xbmc::InfoTagGame"
```

**Comment by garbear** [comment]
What you say makes sense, and my PR isn't the right approach, so I've closed it.

I think I found the original source of the problem. `li.getGameInfoTag()` is creating a new, empty game tag not linked to anything. This differs from the behavior of `li.getVideoInfoTag()`, which should let you set fields on the listitem.

I opened https://github.com/xbmc/xbmc/pull/23076, which makes picture/game infotags behave like video/music infotags. Your original code should now work:

```python
gameinfo = li.getGameInfoTag()
gameinfo.setTitle(info_dict.get('title'))
gameinfo.setPlatform(info_dict.get('platform'))
gameinfo.setGenres(info_dict.get('genres'))
gameinfo.setPublisher(info_dict.get('publisher'))
gameinfo.setOverview(info_dict.get('overview'))
gameinfo.setYear(info_dict.get('year'))
gameinfo.setGameClient(info_dict.get('gameclient'))
```

I probably can't test until after devcon, so if you're able to test, let me know if https://github.com/xbmc/xbmc/pull/23076 solves your problem. I'm launching test builds as we speak.
